src/error.rs

```diff
@@ -19,126 +19,117 @@ pub type Result<T> = std::result::Result<T, Error>;
 #[allow(clippy::large_enum_variant)]
 #[serde(tag = "type", content = "error")]
 pub enum Error {
-    /// Error when building tagged_data blocks
-    #[error("Error when building tagged_data block: {0}")]
-    TaggedDataError(String),
+    /// Bee rest api error
+    #[error("{0}")]
+    #[serde(serialize_with = "display_string")]
+    BeeRestApiError(#[from] bee_api_types::error::Error),
+    /// Blake2b256 Error
+    #[error("{0}")]
+    Blake2b256Error(&'static str),
+    /// Block dtos error
+    #[error("{0}")]
+    #[serde(serialize_with = "display_string")]
+    BlockDtoError(#[from] bee_block::DtoError),
+    /// Block types error
+    #[error("{0}")]
+    #[serde(serialize_with = "display_string")]
+    BlockError(#[from] bee_block::Error),
+    /// The wallet account has enough funds, but split on too many outputs
+    #[error("the wallet account has enough funds, but split on too many outputs: {0}, max. is 128, consolidate them")]
+    ConsolidationRequired(usize),
+    /// Crypto.rs error
+    #[error("{0}")]
+    #[serde(serialize_with = "display_string")]
+    CryptoError(#[from] crypto::Error),
+    /// Address not found
+    #[error("address: {0} not found in range: {1}")]
+    InputAddressNotFound(String, String),
     /// Invalid amount in API response
-    #[error("Invalid amount in API response: {0}")]
+    #[error("invalid amount in API response: {0}")]
     InvalidAmount(String),
-    /// Error when building transaction blocks
-    #[error("Error when building transaction block")]
-    TransactionError,
+    /// Invalid BIP32 chain data
+    #[error("invalid BIP32 chain data")]
+    InvalidBIP32ChainData,
+    /// Invalid mnemonic error
+    #[error("invalid mnemonic {0}")]
+    InvalidMnemonic(String),
+    /// JSON error
+    #[error("{0}")]
+    #[serde(serialize_with = "display_string")]
+    Json(#[from] serde_json::Error),
+    /// Missing input for utxo chain
+    #[error("missing input: {0}")]
+    MissingInput(String),
+    /// Missing required parameters
+    #[error("must provide required parameter: {0}")]
+    MissingParameter(&'static str),
+    /// No input with matching ed25519 address provided
+    #[error("no input with matching ed25519 address provided")]
+    MissingInputWithEd25519Address,
+    /// Error on API request
+    #[error("node error: {0}")]
+    NodeError(String),
+    /// The block doesn't need to be promoted or reattached
+    #[error("block ID `{0}` doesn't need to be promoted or reattached")]
+    NoNeedPromoteOrReattach(String),
+    /// The requested data was not found.
+    #[error("the requested data was not found.")]
+    NotFound,
```

Comment:
@thibault-martinez I don't like that the error doesn't contain any more information, what do you think about NotFound(String) where the String is the url for which we got this response?

Reply:
Yeah good idea 👍🏻